benchmarks/common.py

```diff
@@ -667,53 +688,56 @@ def run_one_model(
     ):
         t0 = time.perf_counter()
         tolerance = self.get_tolerance(is_training, current_device, name)
-        with self.pick_grad(name, is_training):
-            mode = "train" if is_training else "eval"
-            sys.stdout.write(f"{current_device:4} {mode:5} {current_name:34} ")
-            sys.stdout.flush()
-            for submod in itertools.chain([model], model.modules()):
-                assert not torchdynamo.utils.is_jit_model(submod)
-            torch.manual_seed(1337)
-            correct_result = model_iter_fn(
-                copy.deepcopy(model), torchdynamo.utils.clone_inputs(example_inputs)
-            )
-
-            torch.manual_seed(1337)
-            if current_name not in self.non_deterministic_models:
-                correct_rerun_result = model_iter_fn(
+        # TF32 can amplify minor noises. So, disable TF32 for accuracy comparison
+        with no_tf32():
+            with self.pick_grad(name, is_training):
+                mode = "train" if is_training else "eval"
+                sys.stdout.write(f"{current_device:4} {mode:5} {current_name:34} ")
+                sys.stdout.flush()
+                for submod in itertools.chain([model], model.modules()):
+                    assert not torchdynamo.utils.is_jit_model(submod)
+                torch.manual_seed(1337)
+                correct_result = model_iter_fn(
                     copy.deepcopy(model), torchdynamo.utils.clone_inputs(example_inputs)
                 )
-                if not same(correct_result, correct_rerun_result):
-                    print("INCORRECT - Variation in Eager runs itself")
-                    if not skip_accuracy_check:
-                        return sys.exit(-1)
 
-            torch.manual_seed(1337)
-            torchdynamo.reset()
-            if experiment.func is cold_start_experiment:
-                results = []
-                results.append(experiment(model, example_inputs, optimize_ctx))
-                print(" ".join(map(str, results)))
-                return 0
+                torch.manual_seed(1337)
+                if current_name not in self.non_deterministic_models:
+                    correct_rerun_result = model_iter_fn(
+                        copy.deepcopy(model),
+                        torchdynamo.utils.clone_inputs(example_inputs),
+                    )
+                    if not same(correct_result, correct_rerun_result):
+                        print("INCORRECT - Variation in Eager runs itself")
+                        if not skip_accuracy_check:
+                            return sys.exit(-1)
+
+                torch.manual_seed(1337)
+                torchdynamo.reset()
+                if experiment.func is cold_start_experiment:
+                    results = []
+                    results.append(experiment(model, example_inputs, optimize_ctx))
```

Comment:
Won't this disable tf32 for our performance measurements?

Reply:
This one is specific to cold_start_experiment. The real experiment is called later on.
But, I am moving this one out of scope of with no_tf32.